octobot_trading/personal_data/orders/order.py

```diff
@@ -301,7 +339,48 @@ async def on_filled(self):
         """
         Filling complete callback
         """
-        # nothing to do by default
+        await self._trigger_chained_orders()
+
+    async def _trigger_chained_orders(self):
+        logger = logging.get_logger(self.get_logger_name())
+        for index, order in enumerate(self.chained_orders):
+            if order.should_be_created():
+                logger.debug(f"Creating chained order {index + 1}/{len(self.chained_orders)}")
+                await order.create_as_chained_order()
+            else:
+                logger.debug(f"Skipping cancelled chained order {index + 1}/{len(self.chained_orders)}")
+
+    async def create_as_chained_order(self):
+        try:
+            self.is_waiting_for_chained_trigger = False
+            if not self.trader.simulate and self.has_been_bundled:
+                # exchange should have created it already, it will automatically be fetched at the next update
+                # TODO: check if we need to really fetch it through updater etc
+                #  (issue is that we don't have an id yet so we can't just fetch this order in particular)
+                pass
+            else:
+                await self.trader.create_order(
+                    self,
+                    loaded=False,
+                    params=self.exchange_creation_params,
+                    **self.trader_creation_kwargs
+                )
+        finally:
+            # set created now to consider creation failures as created as well (the caller can always retry later on)
+            self.created = True
```

Comment:
I don't think it should be in Order. I don't really like the idea that an Order can trigger an exchange action like this. I think this method should be put somewhere else like in order_util

Reply:
Agree